AtEventDisplay/AtSidebar/AtSidebarIntMacro.h

```diff
@@ -0,0 +1,54 @@
+#ifndef ATSIDEBARINTMACRO_H
+#define ATSIDEBARINTMACRO_H
+
+#include "AtSidebarAddon.h"
+
+#include <Rtypes.h> // for THashConsistencyHolder, UInt_t, ClassDef...
+
+#include <GuiTypes.h> // for Pixel_t
+
+#include <functional>
+#include <string>
+#include <utility> // for move
+
+class TBuffer;
+class TClass;
+class TGWindow;
+class TMemberInspector;
+
+class AtSidebarIntMacro : public AtSidebarAddon {
```

Comment:
Needs doxygen. What does this class do? This feels like one of the sidebar helper things? I think those are all in the same file. Is there a reason for this to not also be there?

Reply:
AtSidebarIntMacro is like AtSidebarInfoMacro. Both of these are generic sidebar items that can be customized. In this case it is an int box that can be used to do arbitrary things. It means we don't need a special class for each thing we might want adjustable parameters for in the viewer, like we have for some of the PSA methods.

Reply:
This might be beyond what you want to do, but what I think this should be replaced with eventually:

Using signals/slots https://root.cern/manual/signal_slot/ it should be possible to add this capability into AtSIdebarAddon where the callback includes the label of the entry that we can use to call the correct MacroFunction. (AtSidebarAddaon would require another `map<String, macropointer>`, but it would let us add multiple without spawning a new class for each one.

Reply:
Right now. we need to instantiate a new class which adds a new sidebar object for each intbox. And it would be nice if we could just do the `AddIntBox(label, ... , std::func<void(int)>)`

Reply:
I like this idea. But it's not happening before the holidays and this pull request has been sitting for long enough, I think.

I might take a look at adding that functionality once I'm back to work in the new year, but as it's own thing.

Reply:
For now I added documentation to the class (and InfoMacro).